### PR TITLE
Improve slow import/export duration

### LINKED ISSUE #79: Slow export duration when exporting large datasets
I am exporting a large table (>1m rows).  The export duration for this is very long, because I am using [django celery backend](https://django-celery-results.readthedocs.io/en/latest/) which writes to the db, and there is an [update to the db for every row in the export](https://github.com/saritasa-nest/django-import-export-extensions/blob/08038894a4833ae9e367d2dcf850b058eee197eb/import_export_extensions/resources.py#L214).

The workaround for this is to override the `update_task_state()` method:

```python
    def update_task_state(
        self,
        state: str,
    ):
        # performance optimization to prevent update of task state for each row
        if state == TaskState.EXPORTING.name:
            return
        super().update_task_state(state)
```

Perhaps we could have an option to disable update during export, or we could have a class variable for `Resource`, which means we update the status every N rows, maybe something like:

```python
    class Meta:
        model = models.Artist
        fields = [
            "id",
            "name",
        ]
        status_update_row_count = 1000
```

### PR DESCRIPTION
**Pull request by Eg0ra**
Closed #79

### CODE REVIEW

File: HISTORY.rst [hunk 1]
```diff
@@ -1,6 +1,12 @@
 =======
 History
 =======
+UNRELEASED
+------------------
+* Fix issue with slow export duration (https://github.com/saritasa-nest/django-import-export-extensions/issues/79):
+
+  * Add setting ``STATUS_UPDATE_ROW_COUNT`` which defines the number of rows after import/export of which the task status is updated;
```
Comment (TheSuperiorStanislav) [inline HISTORY.rst:8]: Add mention of default value

File: import_export_extensions/resources.py [hunk 2]
```diff
@@ -1,14 +1,16 @@
 import collections
 import typing
 from enum import Enum
+from functools import cached_property
```
Comment (TheSuperiorStanislav) [inline import_export_extensions/resources.py:4]:
```suggestion
import enum
import functools
```
For consistency

File: import_export_extensions/resources.py [hunk 3]
```diff
@@ -229,15 +244,15 @@ def initialize_task_state(
             return
 
         if isinstance(queryset, QuerySet):
-            total = queryset.count()
+            self.total_objects_count = queryset.count()
```
Comment (TheSuperiorStanislav) [inline import_export_extensions/resources.py:247]:
```suggestion
            self.total_objects_count = queryset.count() if isinstance(queryset, QuerySet) else len(queryset)
```

File: import_export_extensions/resources.py [hunk 4]
```diff
@@ -254,15 +269,18 @@ def update_task_state(
         if not current_task or current_task.request.called_directly:
```
Comment (TheSuperiorStanislav) [inline import_export_extensions/resources.py:269]: Update docs above

File: import_export_extensions/resources.py [hunk 5]
```diff
@@ -254,15 +269,18 @@ def update_task_state(
...
+        if (
+            self.current_object_number % self.status_update_row_count == 0
+            or self.current_object_number == self.total_objects_count
+        ):
+            self._update_current_task_state(
+                state=state,
+                meta=dict(
+                    current=self.current_object_number,
+                    total=self.total_objects_count,
+                ),
```
Comment (TheSuperiorStanislav) [inline import_export_extensions/resources.py:282]:
```suggestion
                meta={
                    "current": self.current_object_number,
                    "total": self.total_objects_count,
                },
```

File: import_export_extensions/resources.py [hunk 6]
```diff
@@ -194,6 +207,8 @@ def export(
         """Init task state before exporting."""
         if queryset is None:
             queryset = self.get_queryset()
+
+        queryset = self.filter_export(queryset, **kwargs)
```
Comment (TheSuperiorStanislav) [inline import_export_extensions/resources.py:211]: Why did you add it?

Comment (Eg0ra) [reply import_export_extensions/resources.py:213]: If you define `filter_export` in resource class and add any filter, `total_objects_count` will be calculated incorrectly because `filter_export` is called inside parent export method
https://github.com/django-import-export/django-import-export/blob/main/import_export/resources.py#L1098

I added a comment

**Review by TheSuperiorStanislav — APPROVED** [review]

File: import_export_extensions/resources.py [hunk 7]
```diff
@@ -247,22 +266,24 @@ def update_task_state(
...
+            self.current_object_number % self.status_update_row_count == 0
+            or self.current_object_number == self.total_objects_count
+        ):
+            self._update_current_task_state(
+                state=state,
+                meta={
+                    "current": self.current_object_number,
+                    "total": self.total_objects_count,
+                },
+            )
```
Comment (NikAzanov) [inline import_export_extensions/resources.py:286]: I think this condition is complicated, so it requires more documentation: either as comment or docstring

**Review by NikAzanov — APPROVED** [review]

File: import_export_extensions/resources.py (hunk 7 above)
Comment (TheSuperiorStanislav) [reply import_export_extensions/resources.py:295]: Suggest to try break up condition in separate vars to improve readability
```suggestion
        self.current_object_number += 1
        is_reached_update_count = self.current_object_number % self.status_update_row_count == 0
        is_last_object = self.current_object_number == self.total_objects_count
        if is_reached_update_count or is_last_object:
            self._update_current_task_state(
                state=state,
                meta={
                    "current": self.current_object_number,
                    "total": self.total_objects_count,
                },
            )
```

File: import_export_extensions/resources.py [hunk 8]
```diff
@@ -247,22 +266,36 @@ def update_task_state(
     ):
         """Update state of the current event.
 
-        Receives meta of the current task and increase the `current`
-        field by 1.
+        Receives meta of the current task and increase the `current`. Task
+        state is updated when current item is a multiple of
+        `self.status_update_row_count` or equal to  total number of items.
```
Comment (OttoAndrey) [inline import_export_extensions/resources.py:271]: Extra space
```suggestion
        `self.status_update_row_count` or equal to total number of items.
```

**Review by OttoAndrey — APPROVED** [review]
